doozer/doozerlib/backend/rebaser.py

```
@@ -0,0 +1,1549 @@
+                          version: str, release: str, downstream_parents: Optional[List[str]], force_yum_updates: bool):
+        with exectools.Dir(dest_dir):
+            self._generate_repo_conf(metadata, dest_dir, self._runtime.repos)
+
+            self._generate_config_digest(metadata, dest_dir)
+
+            self._write_cvp_owners(metadata, dest_dir)
+
+            self._write_fetch_artifacts(metadata, dest_dir)
+
+            self._write_osbs_image_config(metadata, dest_dir, source, version)
+
+            df_path = dest_dir.joinpath('Dockerfile')
+            self._update_dockerfile(metadata, source, df_path, version, release, downstream_parents, force_yum_updates)
+
+            self._update_csv(metadata, dest_dir, version, release)
+
+            return version, release
+
+    def _make_actual_release_string(self, metadata: ImageMetadata, input_release: str, private_fix: bool, source: Optional[SourceResolution]) -> str:
+        """ Given a input_release string (may contain .p?), make an actual release string.
+
+        e.g. 4.17.0-202407241200.p? -> 4.17.0-202407241200.p0.assembly.stream.gdeadbee.el9
+        """
+        sb = io.StringIO()
+        if input_release.endswith(".p?"):
+            sb.write(input_release[:-3])  # strip .p?
+            pval = ".p1" if private_fix else ".p0"
+            sb.write(pval)
+        elif self._runtime.group_config.public_upstreams:
+            raise ValueError(f"'release' must end with '.p?' for an image with a public upstream but its actual value is {input_release}")
+
+        if source and source.commit_hash:
+            sb.write(".g")
+            sb.write(source.commit_hash_short)
+
+        if self._runtime.assembly:
+            sb.write(".assembly.")
+            sb.write(self._runtime.assembly)
+
+        el_ver = 0
+        try:
+            el_ver = metadata.branch_el_target()
+        except ValueError:
+            pass
+        if el_ver:
```

We've grown to depend on `.elX` suffix in our nvr names, so i think we should error out if it is not found instead of passing, so we can investigate

---

but can also do in a separate PR, so upto you if you wanna make this change here

---

Right. This part was taken from existing code and I didn't change it. Let's make a change in future PRs.